jestSetup.js

```diff
@@ -2,3 +2,7 @@ jest.mock('./lib/storage');
 jest.mock('./lib/storage/NativeStorage', () => require('./lib/storage/__mocks__'));
 jest.mock('./lib/storage/WebStorage', () => require('./lib/storage/__mocks__'));
 jest.mock('./lib/storage/providers/LocalForage', () => require('./lib/storage/__mocks__'));
+
+// Jest doesn't recognize native file extensions such as `.web.js` or `.native.js`.
+// This is why we need to default one of the platforms implementations
```

Comment:
The only reason why this is needed, is because jest will not recognise these native file extension. In the end, we can only use the localForageMock in `./src/storage/__mocks__/index.js` for testing..

Reply:
The need for a platform (storage provider) specific is the essential thing for this issue. We need to be able to pass only the delta to the `SQLiteStorage` and pass the (already) modified/merged data to `LocalForage`.

The `Storage.merge(key, changes, modifiedData)` signature with the additional `changes` paremeter solves this, as we can use the changes on native platforms and ignore the `modifiedData`. On web, we would just ignore the `changes`.